baseplate/clients/thrift.py

```diff
@@ -166,6 +167,15 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
                     if span.flags:
                         prot.trans.set_header(b"Flags", str(span.flags).encode())
 
+                    min_timeout = value or None
+                    if self.pool.timeout:
+                        if not min_timeout or self.pool.timeout < min_timeout:
+                            min_timeout = self.pool.timeout
+                    if min_timeout and min_timeout > 0:
+                        prot.trans.set_header(
+                            b"Deadline-Budget", str(int(ceil(min_timeout * 1000))).encode()
```

Comment:
the whole `str(int(ceil(min_timeout * 1000))).encode()` part could be hard to follow for future engineers reading the code, so it's probably worth adding a comment to explain that we are converting it to milliseconds rounding up.